lib/extensions.js

```diff
@@ -162,10 +162,23 @@ module.exports = function(realmConstructor, context) {
             Object.keys(this)
                 .concat(Object.keys(Object.getPrototypeOf(this)))
                 .forEach(key => {
-                    const value = this[key];
-                    result[key] = value instanceof realmConstructor.Object || value instanceof realmConstructor.Collection
-                        ? value.toJSON(key, cache)
-                        : value;
+                    const org = this[key];
+                    let val = undefined;
+
+                    // skip any internals, functions & constructors (in case of class models).
+                    if (key === "_realm" || typeof existing === "function") {
+                      return;
+                    }
+
+                    if (org instanceof realmConstructor.Object || org instanceof realmConstructor.Collection) {
+                        // recursively trigger `toJSON` for Realm instances with the same cache.
+                        val = org.toJSON(key, cache);
+                    } else if (org instanceof ArrayBuffer) {
+                        // convert "data" to base64 strings.
+                        val = Buffer.from(org).toString("base64");
+                    }
+
+                    result[key] = val !== undefined ? val : org;
```

Comment:
I think this change could make us a bit more robust to the global being swapped (as some Jest environments turns out to do).

```suggestion
                    result[key] = typeof val !== "undefined" ? val : org;
```